engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java

```diff
@@ -353,6 +354,109 @@ public void binaryWithVec() throws Exception {
     assertEquals("binary.Bin", ok3.getMetaObject().getMetaQualifiedName());
   }
 
+  @Test
+  public void partiallyAppliedConstructor() throws Exception {
+    final URI uri = new URI("memory://partial.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import Integer
+
+    type V
+        Val a b c
+
+    create x:V = x.a + x.b + x.c
+
+    mix a =
+      partial = V.Val 1 a
+      create partial
+    """, uri.getHost())
+            .uri(uri)
+            .buildLiteral();
+
+    var module = ctx.eval(src);
+    var mix = module.invokeMember("eval_expression", "mix");
+
+    try {
+      var res = mix.execute(7);
+      fail("No result expected: " + res);
+    } catch (PolyglotException ex) {
+      assertContains("Type error", ex.getMessage());
+      assertContains("expected `x` to be V", ex.getMessage());
+      assertContains("got V.Val[partial", ex.getMessage());
+      assertContains("a=1", ex.getMessage());
+      assertContains("b=7", ex.getMessage());
+      assertContains("c=_", ex.getMessage());
+    }
+  }
+
+  @Test
+  public void oversaturatedFunction() throws Exception {
+    final URI uri = new URI("memory://oversaturated.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import Integer
+
+    fn a b c =
+      sum = a + b + c
+      add a = sum + a
+      add
+
+    neg x:Integer = -x
+
+    mix n = neg (fn 2 a=4 n)
+    """, uri.getHost())
+            .uri(uri)
+            .buildLiteral();
+
+    var module = ctx.eval(src);
+    var mix = module.invokeMember("eval_expression", "mix");
+
+    try {
+      var res = mix.execute(7);
+      fail("No result expected: " + res);
+    } catch (PolyglotException ex) {
+      assertContains("Type error", ex.getMessage());
+      assertContains("expected `x` to be Integer", ex.getMessage());
+      assertContains("got oversaturated.fn[", ex.getMessage());
+      assertContains("a=2", ex.getMessage());
+      assertContains("b=7", ex.getMessage());
+      assertContains("c=_", ex.getMessage());
+      assertContains("+a=4", ex.getMessage());
+    }
+  }
+
+  @Test
+  public void suspendedArgumentsUnappliedFunction() throws Exception {
+    final URI uri = new URI("memory://suspended.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import Integer
+
+    fn ~a ~b ~c =
+      add x = if x == 0 then 0 else x * (a + b + c)
+      add
+
+    neg x:Integer = -x
+
+    mix a = neg (fn c=(2/0) b=(a/0))
+    """, uri.getHost())
+            .uri(uri)
+            .buildLiteral();
+
+    var module = ctx.eval(src);
+    var mix = module.invokeMember("eval_expression", "mix");
+
+    try {
+      var res = mix.execute(0);
+      fail("No result expected: " + res);
+    } catch (PolyglotException ex) {
+      assertContains("Type error", ex.getMessage());
```

Comment:
Yields following single line message:
```
Type error: expected `x` to be Integer, but got 
suspended.fn[suspended:3-5] 
self=suspended 
a=_ 
b=suspended.mix<arg-b>[suspended:9:28-30] 
c=suspended.mix<arg-c>[suspended:9:20-22].
```
I assume we want _single line_ messages although the description of arguments may be pretty looong.